### ISSUE TITLE
[Bug]: Sampling parameter fixed issue while doing speculative sampling verification step

### ISSUE THREAD

**Issue by jeongin601** [vllm-project/vllm#9834]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Your output of `python collect_env.py` here
```

PyTorch version: 2.4.0+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.27.9
Libc version: glibc-2.35

Python version: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.4.239-1.el7.elrepo.x86_64-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.3.107
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA A100-SXM4-80GB
GPU 1: NVIDIA A100-SXM4-80GB
GPU 2: NVIDIA A100-SXM4-80GB
GPU 3: NVIDIA A100-SXM4-80GB
GPU 4: NVIDIA A100-SXM4-80GB
GPU 5: NVIDIA A100-SXM4-80GB
GPU 6: NVIDIA A100-SXM4-80GB
GPU 7: NVIDIA A100-SXM4-80GB

Nvidia driver version: 535.54.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.7
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   43 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          64
On-line CPU(s) list:             0-63
Vendor ID:                       AuthenticAMD
Model name:                      AMD EPYC 7302 16-Core Processor
CPU family:                      23
Model:                           49
Thread(s) per core:              2
Core(s) per socket:              16
Socket(s):                       2
Stepping:                        0
BogoMIPS:                        5989.21
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate sme ssbd mba sev ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif umip rdpid overflow_recov succor smca
Virtualization:                  AMD-V
L1d cache:                       1 MiB (32 instances)
L1i cache:                       1 MiB (32 instances)
L2 cache:                        16 MiB (32 instances)
L3 cache:                        256 MiB (16 instances)
NUMA node(s):                    2
NUMA node0 CPU(s):               0-15,32-47
NUMA node1 CPU(s):               16-31,48-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Vulnerable
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:        Vulnerable, IBPB: disabled, STIBP: disabled, PBRSB-eIBRS: Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.24.4
[pip3] nvidia-cublas-cu12==12.1.3.1
[pip3] nvidia-cuda-cupti-cu12==12.1.105
[pip3] nvidia-cuda-nvrtc-cu12==12.1.105
[pip3] nvidia-cuda-runtime-cu12==12.1.105
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.0.2.54
[pip3] nvidia-curand-cu12==10.3.2.106
[pip3] nvidia-cusolver-cu12==11.4.5.107
[pip3] nvidia-cusparse-cu12==12.1.0.106
[pip3] nvidia-dali-cuda120==1.32.0
[pip3] nvidia-ml-py==12.560.30
[pip3] nvidia-nccl-cu12==2.20.5
[pip3] nvidia-nvjitlink-cu12==12.6.77
[pip3] nvidia-nvtx-cu12==12.1.105
[pip3] nvidia-pyindex==1.0.9
[pip3] onnx==1.15.0rc2
[pip3] optree==0.10.0
[pip3] pynvml==11.4.1
[pip3] pytorch-quantization==2.1.2
[pip3] pyzmq==25.1.2
[pip3] torch==2.4.0
[pip3] torch-tensorrt==2.2.0a0
[pip3] torchdata==0.7.0a0
[pip3] torchtext==0.17.0a0
[pip3] torchvision==0.19.0
[pip3] transformers==4.39.3
[pip3] triton==3.0.0
[conda] Could not collect
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.5.4
vLLM Build Flags:
CUDA Archs: 5.2 6.0 6.1 7.0 7.2 7.5 8.0 8.6 8.7 9.0+PTX; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    NIC0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV12    NV12    NV12    NV12    NV12    NV12    NV12    NODE    0-15,32-47      0         N/A
GPU1    NV12     X      NV12    NV12    NV12    NV12    NV12    NV12    NODE    0-15,32-47      0         N/A
GPU2    NV12    NV12     X      NV12    NV12    NV12    NV12    NV12    PXB     0-15,32-47      0         N/A
GPU3    NV12    NV12    NV12     X      NV12    NV12    NV12    NV12    PXB     0-15,32-47      0         N/A
GPU4    NV12    NV12    NV12    NV12     X      NV12    NV12    NV12    SYS     16-31,48-63     1         N/A
GPU5    NV12    NV12    NV12    NV12    NV12     X      NV12    NV12    SYS     16-31,48-63     1         N/A
GPU6    NV12    NV12    NV12    NV12    NV12    NV12     X      NV12    SYS     16-31,48-63     1         N/A
GPU7    NV12    NV12    NV12    NV12    NV12    NV12    NV12     X      SYS     16-31,48-63     1         N/A
NIC0    NODE    NODE    PXB     PXB     SYS     SYS     SYS     SYS      X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0

</details>

### 🐛 Describe the bug

**Example code**

1. server
`python vllm/vllm/entrypoints/openai/api_server.py -tp 4 --model meta-llama/Llama-2-70b-hf --port 8000 --gpu-memory-utilization 0.8 --trust-remote-code --speculative-model meta-llama/Llama-2-7b-hf --use-v2-block-manager --num_speculative_tokens 1`

2. completion
```
openai_api_base = "http://0.0.0.0:8041/v1"
client = OpenAI(base_url=openai_api_base,)

models = client.models.list()
model = models.data[0].id

completion = client.completions.create(
        model=model,
        prompt="Hello, my name is",
        echo=False,
        n=1,
        stream=False,
        temperature=0.6,
        top_p=0.6,
        max_tokens=128,
)
print(completion)
```

**Bug**
- Setting
  I am running **speculative sampling** with num_speculative_tokens = 1.
In this setting, verification with target model should sample two logits in parallel.

- Problem
  - The bug here is that when I print out the top_p value of sampling parameters, 
**the first top_p is always fixed to 1,** 
which affects the acceptance of the token.
(FYI, same with the example above, I used top_p as 0.6)
  - <img src="https://github.com/user-attachments/assets/cee62039-77bf-4af1-b795-41ad2610d9f6" width="70%" height="70%">

  -  It is also same when the num_speculative_tokens value is different.
 The first top_p value is always fixed to 1.
 
 - The first top_p value should also be set to the specific sampling parameter value that is given as input.

- Test setting 
  - sampling param: top_p = 0.9 / temperature = 0.6
  - Target model / Draft model : LLaMA3-70B / LLaMA3-8B

- As-Is 
  - sampling param: top_p = 0.9
  - Speculative metrics: Draft acceptance rate: 0.651, System efficiency: 0.825, Number of speculative tokens: 1, Number of accepted tokens: 13496, Number of draft tokens: 20747, Number of emitted tokens: 34243.     
  - Draft probs are filtered, and target probs are not filtered when comparing the probabilities.  
  - <img width="478" alt="스크린샷 2024-11-10 오후 8 54 56" src="https://github.com/user-attachments/assets/2326722d-a648-480a-ba0c-de83bc2b0329">

- Modified 
  -  Speculative metrics: Draft acceptance rate: 0.819, System efficiency: 0.910, Number of speculative tokens: 1, Number of accepted tokens: 15311, Number of draft tokens: 18694, Number of emitted tokens: 34005.

   - Both draft probs and target probs are filtered  
   - <img width="436" alt="스크린샷 2024-11-10 오후 10 03 52" src="https://github.com/user-attachments/assets/5d23c3a2-181e-4ee6-a5cd-fd281d12de57">

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
